src/aics-image-viewer/components/App/index.tsx

```diff
@@ -135,9 +136,10 @@ const defaultProps: AppProps = {
 };
 
 /** A `useState` that also creates a getter function for breaking through closures */
-function useStateWithGetter<T>(initialState: T): [T, (value: T) => void, () => T] {
+function useStateWithGetter<T>(initialState: T | (() => T)): [T, (value: T) => void, () => T] {
   const [state, setState] = useState(initialState);
-  const stateRef = useRef(initialState);
+  const isSetterFunction = (val: T | (() => T)): val is () => T => typeof val === "function";
+  const stateRef = useRef(isSetterFunction(initialState) ? initialState() : initialState);
```

Comment:
this useStateWithGetter function is a bit cryptic to me and lines like 141-142 hard to parse.  I don't know if there is a more readable formulation of line 141.  
Maybe a couple of more verbose comments saying what the intent is, and what is the difference between the `T` version and the `(()=>T)` version?

Reply:
Yeah I got weird with this one. On a second look there's something far simpler I could have done to support what I was trying to support. (Which is the convention that, if the argument to `useState` is a function, that function is assumed to hold an expensive calculation of the initial value of that state, rather than itself being the initial value of state.)